1.Marginal-Gain-Multiple-Mutations.py

```diff
@@ -0,0 +1,257 @@
+        for fold, score in enumerate(grid_score.cv_validation_scores):
+            row = grid_score.parameters.copy()
+            row['fold'] = fold
+            row['score'] = score
+            rows.append(row)
+    df = pd.DataFrame(rows)
+    return df
+
+def get_coeffs(pipeline, X_train, model_type):
+    """
+    Get a dataframe with the training coefficients 
+    """
+    final_pipeline = pipeline.best_estimator_
+    final_classifier = final_pipeline.named_steps['classify']
+    
+    # Get indices of features
+    if model_type == 'model a': select_step = 'select'
+    else: select_step = 'features'
+    select_indices = final_pipeline.named_steps[select_step].transform(
+        np.arange(len(X_train.columns)).reshape(1, -1)
+    ).tolist()
+    if model_type =='model b': select_indices = [x for sublist in select_indices for x in sublist]
+    
+    # Make df features, weights
+    coef_df = pd.DataFrame.from_items([
+        ('feature', X_train.columns[select_indices]),
+        ('weight', final_classifier.coef_[0]),
+    ])
+
+    return coef_df
+
+def get_sign_coeffs(coef_df):
+    return (coef_df.weight>0).sum(), (coef_df.weight<0).sum()
+
+def get_ranks_covariates_feat(coef_df):
+    coef_df['abs'] = coef_df['weight'].abs()
+    coef_df = coef_df.sort_values('abs', ascending=False)
+    
+    def RepresentsInt(s):
+        try: 
+            int(s)
+            return True
+        except ValueError:
+            return False
+
+    coef_df['is_cov_feat'] = [not RepresentsInt(x) for x in coef_df['feature']]
+    ranks = np.flatnonzero(coef_df['is_cov_feat'])
+    return ranks
+
+auroc_dfs = {}
+
+
+# ## Train the models.
+
+# In[11]:
+
+get_ipython().run_cell_magic('time', '', '# Train model a: covariates only.\nwarnings.filterwarnings("ignore") # ignore deprecation warning for grid_scores_\nrows = list()\nfor m in list(mutations):\n    series = pd.Series()\n    series[\'mutation\'] = m\n    series[\'symbol\'] = mutations[m]\n    rows.append(get_aurocs(X[\'model a\'], Y[m], cv_pipeline[\'model a\'], series, \'model a\'))\nauroc_dfs[\'model a\'] = pd.DataFrame(rows)\nauroc_dfs[\'model a\'].sort_values([\'symbol\', \'testing_auroc\'], ascending=[True, False], inplace=True)')
+
+
+# In[12]:
+
+get_ipython().run_cell_magic('time', '', '# Train model b: covariates with gene expression data.\nwarnings.filterwarnings("ignore") # ignore deprecation warning for grid_scores_\nrows = list()\nfor m in list(mutations):\n    series = pd.Series()\n    series[\'mutation\'] = m\n    series[\'symbol\'] = mutations[m]\n    rows.append(get_aurocs(X[\'model b\'], Y[m], cv_pipeline[\'model b\'], series, \'model b\'))\nauroc_dfs[\'model b\'] = pd.DataFrame(rows)\nauroc_dfs[\'model b\'].sort_values([\'symbol\', \'testing_auroc\'], ascending=[True, False], inplace=True)')
+
+
+# In[18]:
+
+display(auroc_dfs['model a'])
+display(auroc_dfs['model b'])
+
+
+# In[14]:
+
+auroc_dfs['model a'].to_pickle('./covariates_only.pkl')
```

Comment:
Any reason for the switch to pickles over TSV here -- TSV is preferable so its readable outside of Python. 

Consider combining both tables into a single dataframe for storage by adding a `model` column and then concatenating.